tests/test_asyncio.py

```diff
@@ -30,26 +30,74 @@ async def test_asyncio_bad_put_raises(dummy_server_async):
     assert dummy_server_async.received == ["PCAP.thing=1"]
 
 
-@pytest.mark.asyncio
+def mocked_clear():
+    # rather than unsetting the event in pandablocks.asyncio we need to mock
+    # so that the user is setting it every 3 seconds
+    asyncio.get_running_loop().call_soon_threadsafe(asyncio.sleep(1))
+
+
 @pytest.mark.parametrize("disarmed", [True, False])
 @pytest.mark.parametrize("flush_period", [0.1, None])
-async def test_asyncio_data(
+async def test_asyncio_data_periodic_flushing(
     dummy_server_async, fast_dump, fast_dump_expected, disarmed, flush_period
 ):
     if not disarmed:
         # simulate getting the data without the END marker as if arm was not pressed
         fast_dump = (x.split(b"END")[0] for x in fast_dump)
         fast_dump_expected = list(fast_dump_expected)[:-1]
+
     dummy_server_async.data = fast_dump
     events = []
     async with AsyncioClient("localhost") as client:
         async for data in client.data(frame_timeout=1, flush_period=flush_period):
             events.append(data)
+
             if len(events) == len(fast_dump_expected):
                 break
     assert fast_dump_expected == events
 
 
+async def test_asyncio_data_manual_flushing(dummy_server_async, fast_dump):
```

Comment:
It looks a little odd to have a test without any `assert` statements. All this will test as-is is that there were no exceptions thrown. We should test that we get back the data we expect?

Reply:
The idea behind the test was to test the button click causes a flush - if it didn't then the `asyncio.gather` statement would timeout and the test would fail. I'm happy to also check the data.